inputs/gin/parseFunction.go

```diff
@@ -477,13 +483,15 @@ func parseFunction(s *gengo.Service, log zerolog.Logger, currRoute *gengo.Route,
 	return nil
 }
 
-func parseFromCalledFunction(log zerolog.Logger, callExpr *ast.CallExpr, argNo int, pkgName, pkgPath string, imports []*ast.ImportSpec, onExtract func(result astUtils.ParseResult)) (error, bool) {
+// parseFromCalledFunction parses a function call that an argument is a reference to something
+// It returns true if a new pass is needed to parse the function (i.e. it has had to extract another different type)
+func parseFromCalledFunction(s *gengo.Service, log zerolog.Logger, callExpr *ast.CallExpr, argNo int, pkgName, pkgPath, workDir string, imports []*ast.ImportSpec, onExtract func(result astUtils.ParseResult)) (error, bool) {
```

Comment:
This function has unnecessary argument as they are contained in gengo.Service

Example of new function argument:

```func parseFromCalledFunction(s *gengo.Service, callExpr *ast.CallExpr, argNo int, pkgName, pkgPath string, imports []*ast.ImportSpec, onExtract func(result astUtils.ParseResult)) (error, bool)```

Reply:
Agree with not doing `workDir`, but for now tabling the zerolog.Logger due to requirement for custom logger with custom variables.

Reply:
Agree with not doing `workDir`, but for now tabling the zerolog.Logger due to requirement for custom logger with custom variables.